### ISSUE TITLE
Useful plugins and widgets to include in the package

### ISSUE THREAD

**Issue by MrKarlDilkington** [ExchangeCore/Concrete5-CKEditor#18]
I went through all the CKEditor core team made plugins and found what I think were the most useful and interesting.

**Stylesheet Parser**

This plugin fills the Styles drop-down list with a list of element styles built from the contents.css file. Thanks to this the editor content styles are consistent with the document stylesheet.
(it makes the style dropdown text a preview of the custom style)
http://ckeditor.com/addon/stylesheetparser

**Enhanced Image**
- Integrated image captions.
- An option to center images.
- Dynamic "click and drag" resizing of images.  

http://ckeditor.com/addon/image2

**Table Resize**
This plugin adds support for table column resizing with your mouse. Hover your mouse over the column border to see the cursor change to indicate that the column can be resized. Click and drag your mouse to set the desired column width.
http://ckeditor.com/addon/tableresize

**Media Embed**
This plugin lets you embed media resources directly in the editor.

The plugin introduces a widget that allows you to embed resources (videos, images, tweets, etc.) hosted by other services (called the "content providers") in your content. By default the plugin uses the Iframely proxy service which supports over 1715 content providers such as YouTube, Vimeo, Twitter, Instagram, Imgur, GitHub, or Google Maps.
http://ckeditor.com/addon/embed

**Code Snippet**
This plugin lets you insert rich code snippets with syntax highlighting into the editor.
http://ckeditor.com/addon/codesnippet

**Widget**
This plugin introduces the Widget API. It is required by all plugins that create widgets.

Widgets let you standardize, manage and control the layout and style of specific sections of your WYSIWYG area, all this at the press of a button. They group various elements together so you can easily move widgets around the surrounding content.
http://ckeditor.com/addon/widget

**Placeholder**
This plugin lets you create and edit placeholders (non-editable text fragments). Editing the placeholder text is only possible through the Placeholder Properties dialog window that opens when you double-click an existing placeholder or use the Placeholder toolbar button.

Dependencies: Requires the following plugins to work: Widget, Line Utilities,Dialog, Clipboard.
http://ckeditor.com/addon/placeholder

**Comment by joe-meyer** [comment]
I'm good with doing a build for these and providing them. 

The `Widget` one should probably be included and on by default and not able to be disabled to provide the full functionality for plugin developers who want to hook in on concrete5. That or we need to make a more robust plugin manager that is able to handle dependencies so that way we can minimize the asset footprint on load.

Do you think any others of these should be on by default? The table resize seems like it would be super useful to me, same with enhanced image and possibly stylesheet parser.

**Comment by MrKarlDilkington** [comment]
I agree, I think Enhanced Image, Table Resize, Stylesheet Parser, and Widget should be on by default.

**Comment by joe-meyer** [comment]
Stylesheet parser didn't work well at all so I'm not including it at this point, we can revisit that a bit later in a new issue if need be. There were entirely too many styles being picked up that just weren't meaningful.
![image](https://cloud.githubusercontent.com/assets/1292766/10721142/7cc47800-7b75-11e5-907a-8497161be8b4.png)

Widget is included in the core so if a plugin has it as a dependency it's automatically going to be loaded.

I've included Code Snippet but it's not functioning properly, seems to be stripping out line breaks. Will create a new bug report for this and investigate.

Table resize has been included as a new default

Enhanced Image has been included as a new default

Media Embed has been added as an optional plugin

Completed in c7f766f0e0fca67426b887d5de895232bef4975c
